packages/collaboration-extension/src/index.ts

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+/**
+ * @packageDocumentation
+ * @module collaboration-extension
+ */
+
+import * as Y from 'yjs';
+import { Awareness } from 'y-protocols/awareness';
+import { WebsocketProvider } from 'y-websocket';
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { PageConfig } from '@jupyterlab/coreutils';
+import { DOMUtils } from '@jupyterlab/apputils';
+import {
+  CollaboratorsPanel,
+  ICurrentUser,
+  IGlobalAwareness,
+  IUserMenu,
+  IUserPanel,
+  RendererUserMenu,
+  RTCPanel,
+  User,
+  UserInfoPanel,
+  UserMenu
+} from '@jupyterlab/collaboration';
+import { usersIcon } from '@jupyterlab/ui-components';
+import { AccordionPanel, Menu, MenuBar } from '@lumino/widgets';
+import { URLExt } from '@jupyterlab/coreutils';
+import { ServerConnection } from '@jupyterlab/services';
+import { IStateDB, StateDB } from '@jupyterlab/statedb';
+
+/**
+ * Jupyter plugin providing the ICurrentUser.
+ */
+const userPlugin: JupyterFrontEndPlugin<ICurrentUser> = {
+  id: '@jupyterlab/collaboration-extension:user',
+  autoStart: true,
+  provides: ICurrentUser,
+  activate: (app: JupyterFrontEnd): ICurrentUser => {
+    return new User();
+  }
+};
+
+/**
+ * Jupyter plugin providing the IUserMenu.
+ */
+const userMenuPlugin: JupyterFrontEndPlugin<IUserMenu> = {
+  id: '@jupyterlab/collaboration-extension:userMenu',
+  autoStart: true,
+  requires: [ICurrentUser],
+  provides: IUserMenu,
+  activate: (app: JupyterFrontEnd, user: ICurrentUser): IUserMenu => {
+    const { commands } = app;
+    return new UserMenu({ commands, user });
+  }
+};
+
+/**
+ * Jupyter plugin adding the IUserMenu to the menu bar if collaborative flag enabled.
+ */
+const menuBarPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/collaboration-extension:userMenuBar',
+  autoStart: true,
+  requires: [ICurrentUser, IUserMenu],
+  activate: (
+    app: JupyterFrontEnd,
+    user: ICurrentUser,
+    menu: IUserMenu
+  ): void => {
+    const { shell } = app;
+
+    if (PageConfig.getOption('collaborative') !== 'true') {
+      return;
+    }
+
+    const menuBar = new MenuBar({
+      forceItemsPosition: {
+        forceX: false,
+        forceY: false
+      },
+      renderer: new RendererUserMenu(user)
+    });
+    menuBar.id = 'jp-UserMenu';
+    user.changed.connect(() => menuBar.update());
+    menuBar.addMenu(menu as Menu);
+    shell.add(menuBar, 'top', { rank: 1000 });
+  }
+};
+
+/**
+ * Jupyter plugin creating a global awareness for RTC.
+ */
+const rtcGlobalAwarenessPlugin: JupyterFrontEndPlugin<Awareness | null> = {
```

Comment:
Not sure this plugin should provide a `Awareness | null`. Other plugins requiring `IGlobalAwareness` will expect the service to be available. It might be a bit odd if they have to check whether it is `null` before using it.